src/libstd/path.rs

```diff
@@ -999,16 +1036,58 @@ impl PathBuf {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<P: AsPath> iter::FromIterator<P> for PathBuf {
+impl<'a> From<&'a str> for PathBuf {
```

Comment:
Similarly to the `OsString` case, these may be better expressed via `Into`.

It's also a little sad that the same set of impls as `From<T> for OsString` needs to be listed here, but not necessarily the end of the world.